src/uct/ib/rc/base/rc_iface.c

```diff
@@ -380,19 +392,157 @@ static void uct_rc_iface_tx_ops_cleanup(uct_rc_iface_t *iface)
     ucs_free(iface->tx.ops_buffer);
 }
 
+#if IBV_EXP_HW_TM
+static void uct_rc_iface_release_desc(uct_recv_desc_t *self, void *desc)
+{
+    uct_rc_iface_release_desc_t *release = ucs_derived_of(self,
+                                                          uct_rc_iface_release_desc_t);
+    void *ib_desc = desc - release->offset;
```

Comment:
Address arithmetic with `void *` is UB by C standard, in spite of gcc defines `void *` as `char *` for that.
I noticed below that this code was moved as is from different place, but it's easy to fix because of locality.

Reply:
agree with comment, but we have lots of other places with void arithmetic. Maybe it is worth to declare that in coding style/rules? @yosefe

Reply:
or even add new macro definitions for this. code with multiple type-casting is not good readable

Reply:
too many void* arith in the code to change it now..

Reply:
i don't think this is a point don't do it

Reply:
Opened #1982 for that. In this PR I'll cast void* to char* in this place